app/src/main/java/dev/aurakai/auraframefx/oracledrive/genesis/ai/VertexAIModule.kt

```
@@ -61,7 +82,23 @@ object VertexAIModule {
         securityContext: SecurityContext,
         logger: AuraFxLogger
     ): VertexAIClient {
-        return VertexAIClientImpl()
+        // Get API key from BuildConfig (requires setup in build.gradle.kts)
+        val apiKey = try {
+            BuildConfig.GEMINI_API_KEY.takeIf { it.isNotBlank() }
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)